astroid/brain/brain_builtin_inference.py

```diff
@@ -212,7 +217,7 @@ def _transform_wrapper(node, context: InferenceContext | None = None):
 
     AstroidManager().register_transform(
         nodes.Call,
-        inference_tip(_transform_wrapper),
+        inference_tip(cast(InferFn, _transform_wrapper)),
```

Comment:
Are we not able to just change the signature instead of casting? Should be relatively straightforward?

Reply:
Signatures are fine (well, other than two of them omitting **kwargs, but that didn't seem to be the issue), so I didn't find it straightforward (😅), but perhaps you can help!

```
astroid/brain/brain_builtin_inference.py:219: error: Argument 1 to "inference_tip" has incompatible type "Callable[[Any, InferenceContext | None], Any]"; expected "InferFn[<nothing>]"  [arg-type]
astroid/brain/brain_builtin_inference.py:1080: error: Argument 1 to "inference_tip" has incompatible type "Callable[[Any, InferenceContext | None], Any]"; expected "InferFn[<nothing>]"  [arg-type]
astroid/brain/brain_builtin_inference.py:1086: error: Argument 1 to "inference_tip" has incompatible type "Callable[[Call, InferenceContext | None, KwArg(Any)], Iterator[NodeNG | UninferableBase | Proxy]]"; expected "InferFn[<nothing>]"  [arg-type]
astroid/brain/brain_builtin_inference.py:1093: error: Argument 1 to "inference_tip" has incompatible type "Callable[[Call, InferenceContext | None, KwArg(Any)], Iterator[astroid.nodes.node_classes.List | astroid.nodes.node_classes.Set | astroid.nodes.node_classes.Tuple | astroid.nodes.node_classes.Dict | Const | EmptyNode | UninferableBase]]"; expected "InferFn[<nothing>]"  [arg-type]
```

I'm thinking it's a mypy issue, but I'm not certain which one.

Reply:
Hold on, the **kwargs did fix two of them, looking into the next two...

Reply:
Couldn't figure it out. There are so many open issues in mypy for unions and generics that I didn't narrow it down to one.

Reply:
Added two disables in 
e223578

Reply:
Discussed below, resolving